v2/car.go

```diff
@@ -41,12 +46,90 @@ type (
 	}
 )
 
-// Size gets the size of Header in number of bytes.
-func (h *Header) Size() int {
-	return HeaderBytesSize
+// WriteTo writes this characteristics to the given writer.
+func (c Characteristics) WriteTo(w io.Writer) (n int64, err error) {
+	wn, err := writeUint64To(w, c.Hi)
+	if err != nil {
+		return
+	}
+	n += wn
+	wn, err = writeUint64To(w, c.Lo)
+	if err != nil {
+		return
+	}
+	n += wn
+	return
 }
 
 // Size gets the size of Characteristics in number of bytes.
-func (c *Characteristics) Size() int {
+func (c Characteristics) Size() uint64 {
 	return CharacteristicsBytesSize
 }
+
+// NewHeader instantiates a new CAR v2 header, given the byte length of a CAR v1.
+func NewHeader(carV1Size uint64) *Header {
```

Comment:
I wouldn't make this New func return a pointer, because the Header type is tiny. It barely takes more space than a slice on a 64-bit GOARCH. It's nice to be consistent with always returning pointers from "new" funcs, but it's even nicer to not allocate when we really don't need to :)